src/app/components/message/Time.tsx

```diff
@@ -1,22 +1,28 @@
 import React from 'react';
 import { Text, as } from 'folds';
 import { timeDayMonYear, timeHourMinute, today, yesterday } from '../../utils/time';
+import { useSetting } from '../../state/hooks/settings';
+import { settingsAtom } from '../../state/settings';
 
 export type TimeProps = {
   compact?: boolean;
   ts: number;
 };
 
 export const Time = as<'span', TimeProps>(({ compact, ts, ...props }, ref) => {
+  const [hour24Clock] = useSetting(settingsAtom, 'hour24Clock');
```

Comment:
Subscribing to `settingsAtom` inside `Time` component can downgrade the perf of our timeline as every `Time` component in every message will be listening to it where more perf oriented approach can be to receive `hour24Clock` as component param and we can subscribe to `settingsAtom` once for all.